src/index.ts

```diff
@@ -86,35 +80,38 @@ function fontawesomeSubset(subset: SubsetOption, outputDir: string, options: Fon
         }
 
         const fontFamily = key as keyof typeof fontMap;
-        const svgFileName = fontMap[fontFamily];
-        const svgFilePath = `node_modules/@fortawesome/fontawesome-${options.package}/webfonts/${svgFileName}.svg`;
+        const ttfFileName = fontMap[fontFamily];
+        const ttfFilePath = `node_modules/@fortawesome/fontawesome-${options.package}/webfonts/${ttfFileName}.ttf`;
 
-        if (!existsSync(svgFilePath)) {
-            console.warn(`Unable to find SVG font file for requested font style '${fontFamily}'. Skipping.`);
+        if (!existsSync(ttfFilePath)) {
+            console.warn(`Unable to find TTF font file for requested font style '${fontFamily}'. Skipping.`);
             continue;
         }
 
-        const svgFile = readFileSync(svgFilePath).toString();
-        const glyphsToRemove = findGlyphsToRemove(svgFile, fontFamily, icons);
-        const svgContentsNew = svgFile.replace(new RegExp(`(<glyph glyph-name="(${glyphsToRemove.join("|")})".*?\\/>)`, "gms"), "").replace(/>\s+</gms, "><");
-        const ttfUtils = svg2ttf(svgContentsNew, {
-            fullname: "FontAwesome " + fontFamily,
-            familyname: "FontAwesome",
-            subfamilyname: fontFamily,
-            ts: 0 // Manually specify empty timestamp for deterministic output
-            // Casting these as @types are out of date for svg2ttf
-        } as FontOptions);
-        const ttf = Buffer.from(ttfUtils.buffer);
-
-        makeDirSync(resolve(outputDir));
-
-        const outputFile = resolve(outputDir, svgFileName);
-
-        writeFileSync(`${outputFile}.svg`, svgContentsNew);
-        writeFileSync(`${outputFile}.ttf`, ttf);
-        writeFileSync(`${outputFile}.eot`, ttf2eot(ttf));
-        writeFileSync(`${outputFile}.woff`, ttf2woff(ttf));
-        writeFileSync(`${outputFile}.woff2`, ttf2woff2(ttf));
+        const fontData = fs.readFileSync(ttfFilePath);
+        let iconSubSetUnicode: string[] = [];
+        icons.forEach((ico) => {
+            if (!(ico in iconMeta)) {
+                console.warn(`Icon '${ico}' is not found in font metadata. Skipping.`);
+            } else {
+                iconSubSetUnicode.push(String.fromCodePoint(parseInt(iconMeta[ico]["unicode"], 16)));
+            }
+        });
+
+        outputTypes.forEach((ftype) => {
+            const subsetBuffer = subsetFont(fontData, iconSubSetUnicode.join(" "), {
+                targetFormat: ftype.targetFormat,
+            });
+
+            subsetBuffer.then((data: any) => {
+                const outputFile = resolve(outputDir, ttfFileName);
+                writeFileSync(outputFile + "." + ftype.fileExt, data);
+                if (ftype.fileExt === "ttf") {
+                    writeFileSync(`${outputFile}.eot`, ttf2eot(data));
+                    writeFileSync(`${outputFile}.woff`, ttf2woff(data));
```

Comment:
Wondering if this is necessary anymore. FontAwesome 6.x only outputs `.ttf` and `.woff2`: https://github.com/FortAwesome/Font-Awesome/tree/6.x/webfonts . It could also maybe be an option to output legacy font formats.

Reply:
Good point, I just left there so there are no other changes for anyone using this app. I'll just leave that unless @omacranger wants it changed.